hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/block/TestBlockManager.java

```diff
@@ -355,10 +364,9 @@ public void testBlockAllocationWithNoAvailablePipelines()
       pipelineManager.finalizeAndDestroyPipeline(pipeline, false);
     }
     Assert.assertEquals(0, pipelineManager.getPipelines(type, factor).size());
-    Assert.assertNotNull(blockManager
+    Assert.assertNull(blockManager
```

Comment:
This TestBlockManager test doesn't use MiniOzoneCluster to do the test.  There is no hearbeat send by datanode. So when BlockManager.allocateBlock send out pipeline creation command, it will never receive the confirmation from datanode.  The new pipeline will always be in ALLOCATED state untill timeout. 
Unit test which leverages MiniOzoneCluster don't have this issue.

Reply:
If we make this change the tests lose their significance. Either we can remove these tests or we can make some changes to the test so that heartbeat can be skipped. 
Best I can think of is adding a handler for the event SCMEvents.DATANODE_COMMAND in the test. The handler can move the pipeline to OPEN state on receiving CreatePipelineCommand in the event. This event is fired by RatisPipelineProvider#create. This should solve our problem.